client/src/components/ClubCreationModal.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from "react";
+import Modal from "react-bootstrap/Modal";
+import Button from "react-bootstrap/Button";
+import { useNavigate } from "react-router-dom";
+import { postData } from "../api";
+import { useAuth } from "../AuthContext";
+
+export default function ClubCreationModal(props) {
+  const [clubName, setClubName] = useState("");
+  const [desc, setDesc] = useState("");
+  const [joinStatus, setJoinStatus] = useState("open");
+  const { user } = useAuth();
+  const navigate = useNavigate();
+
+  const onSubmit = async () => {
+    const userId = user?.id;
+    try {
+      await postData(`clubs/create-club`, {
+        userId,
+        clubName,
+        desc,
+        joinStatus,
+      }).then((res) => {
+        navigate(`/club/${res.id}`); //On successful club creation, navigate user to new club page
+      });
+    } catch {
+      console.error("Error creating club");
```

Comment:
Super tiny thing just for debugging purposes: catch(error) {console.log(error);} should suffice since the response in the event of an error would be catch(error) {console.log(`error creating club :${error}`);} anyways (per createClub():37).

Reply:
Good catch, will fix